packages/expo-cli/src/commands/start.js

```diff
@@ -92,6 +99,48 @@ async function action(projectDir, options) {
   log.nested(chalk.green('Logs for your project will appear below. Press Ctrl+C to exit.'));
 }
 
+async function validateDependenciesVersions(projectDir, exp, pkg) {
+  if (!Versions.gteSdkVersion(exp, '33.0.0')) {
+    return;
+  }
+
+  const bundledNativeModules = await JsonFile.readAsync(
+    ConfigUtils.resolveModule('expo/bundledNativeModules.json', projectDir, exp)
+  );
+  const bundleNativeModulesNames = Object.keys(bundledNativeModules);
+  const projectDependencies = Object.keys(pkg.dependencies);
+
+  const modulesToCheck = intersection(bundleNativeModulesNames, projectDependencies);
+  const incorrectDeps = modulesToCheck.reduce((acc, moduleName) => {
```

Comment:
I'd prefer a `for of` loop to `reduce` here. Especially given that the reducer mutates the array, there's no benefit to it over a simple loop.